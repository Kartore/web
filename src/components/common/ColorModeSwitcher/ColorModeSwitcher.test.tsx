import { describe } from 'vitest';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

import * as Stories from './ColorModeSwitcher.stories.tsx';

const { Default } = composeStories(Stories);

describe('Component: ColorModeSwitcher', () => {
  describe('Snapshot', () => {
    it('Default', () => {
      const { asFragment } = render(<Default />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
