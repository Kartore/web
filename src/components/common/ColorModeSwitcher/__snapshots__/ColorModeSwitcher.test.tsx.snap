// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: ColorModeSwitcher > Snapshot > Default 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-8);
  min-width: var(--chakra-sizes-8);
  font-size: var(--chakra-fontSizes-sm);
  -webkit-padding-start: var(--chakra-space-3);
  padding-inline-start: var(--chakra-space-3);
  -webkit-padding-end: var(--chakra-space-3);
  padding-inline-end: var(--chakra-space-3);
  color: var(--chakra-colors-gray-800);
  padding: var(--chakra-space-1);
}

.emotion-0:focus-visible,
.emotion-0[data-focus-visible] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-0:disabled,
.emotion-0[disabled],
.emotion-0[aria-disabled=true],
.emotion-0[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-0:hover,
.emotion-0[data-hover] {
  background: var(--chakra-colors-gray-100);
}

.emotion-0:hover:disabled,
.emotion-0[data-hover]:disabled,
.emotion-0:hover[disabled],
.emotion-0[data-hover][disabled],
.emotion-0:hover[aria-disabled=true],
.emotion-0[data-hover][aria-disabled=true],
.emotion-0:hover[data-disabled],
.emotion-0[data-hover][data-disabled] {
  background: initial;
}

.emotion-0:active,
.emotion-0[data-active] {
  background: var(--chakra-colors-gray-200);
}

<button
    aria-label="Toggle to Dark"
    class="chakra-button emotion-0"
    type="button"
  >
    <svg
      aria-hidden="true"
      data-slot="icon"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.438 10.148c.19-.425-.321-.787-.748-.601A5.5 5.5 0 0 1 6.453 2.31c.186-.427-.176-.938-.6-.748a6.501 6.501 0 1 0 8.585 8.586Z"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
